integration/container/run_linux_test.go

```diff
@@ -566,3 +575,36 @@ func TestContainerShmSize(t *testing.T) {
 		})
 	}
 }
+
+type legacyCreateRequest struct {
+	containertypes.CreateRequest
+	// Mac Address of the container.
+	//
+	// MacAddress field is deprecated since API v1.44. Use EndpointSettings.MacAddress instead.
+	MacAddress string `json:",omitempty"`
+}
+
+func createLegacyContainer(ctx context.Context, t *testing.T, apiClient client.APIClient, desiredMAC string, ops ...func(*container.TestContainerConfig)) string {
+	t.Helper()
+	config := container.NewTestConfig(ops...)
+	ep := "/v" + apiClient.ClientVersion() + "/containers/create"
+	if config.Name != "" {
+		ep += "?name=" + config.Name
+	}
+	res, _, err := request.Post(ctx, ep, request.Host(apiClient.DaemonHost()), request.JSONBody(&legacyCreateRequest{
+		CreateRequest: containertypes.CreateRequest{
+			Config:           config.Config,
+			HostConfig:       config.HostConfig,
+			NetworkingConfig: config.NetworkingConfig,
+		},
+		MacAddress: desiredMAC,
+	}))
```

Comment:
Same for this code; I think the `request` test-utility is fairly useful; it makes sure we can create API requests without accounting for the client and/or API-types.

That said, there's some amount of ugliness in dealing with these, and perhaps it would be nice to have utilities / functional arguments to also being able to unmarshal the response (without having to do so manually)


To some extent, it's been something I've been considering for the client's "request" utilities (which have some amount of logic w.r.t. API-version and the host to connect to); it'd be nice to export those / make them accessible to make it easier to write tooling for extending the API with custom endpoints (but still being able to share the common setup logic).